plugin/src/main/java/io/jenkins/plugins/forensics/git/reference/GitReferenceRecorder.java

```diff
@@ -70,14 +74,57 @@ public void setSkipUnknownCommits(final boolean skipUnknownCommits) {
     public boolean isSkipUnknownCommits() {
         return skipUnknownCommits;
     }
+    
+    /**
+     * Determines the reference build selection strategy, if
+     *  BEST_MATCH_BUILD: default reference point will be used
+     *  PARENT_COMMIT_BUILD: a reference build, which has built the parent commit of the current built revision
+     *  
+     * @param referenceBuildFinderStrategy         
+     */
+    @DataBoundSetter
+    public void setReferenceBuildFinderStrategy(ReferenceBuildFinderStrategy referenceBuildFinderStrategy) {
+        this.referenceBuildFinderStrategy = referenceBuildFinderStrategy;
+    }
+    
+    public ReferenceBuildFinderStrategy getReferenceBuildFinderStrategy() {
+        return referenceBuildFinderStrategy;
+    }
+    
+    /**
+     * Tries to find a reference build, which built the parent commit of the current build
+     * @param owner
+     *          the current build
+     * @param parentCommit
+     *          the parentCommit-SHA1, which was attached to the current build
+     *          
+     * @return reference build, if exists
+     */
+    private Optional<Run<?, ?>> findParentReferenceBuildFromCommit(final Run<?, ?> owner, 
+            final String parentCommit) {
+        for(Run<?, ?> run = owner.getPreviousBuild(); run != null; run = run.getPreviousBuild()) {
+            GitCommitsRecord record = run.getAction(GitCommitsRecord.class);
+            if(record != null && parentCommit.equals(record.getLatestCommit())) {
```

Comment:
Wouldn't it make more sense to reuse the same matching algorithm (see `getReferencePoint`) that searches for specific commits in the set of reference build commits? We cannot guarantee that the parent build has been recorded as latest commit. It might be somewhere in the middle.